### PR TITLE
Evaluator Update to Scala 2.12.10

### PR DESCRIPTION
**Pull request by kiroco12**
This is an update for the evaluator to Scala 2.12.10. This update include:
 - Exclusions for the library dependencies
 - Updated libraries and removed deprecations
 - Basic and easy to use new evaluator client

### CODE REVIEW

File: project/ProjectPlugin.scala [hunk 1]
```diff
@@ -17,10 +17,20 @@ object ProjectPlugin extends AutoPlugin {
...
   object autoImport {
-    lazy val http4sV = "0.15.7a"
+
+    object V {
+      lazy val http4s      = "0.20.13"
+      lazy val circe       = "0.12.3"
+      lazy val log4s       = "1.7.0"
+      lazy val scalatest   = "3.0.8"
+      lazy val roshttp     = "2.2.4"
+      lazy val slf4jSimple = "1.7.28"
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: 1.7.29

File: project/ProjectPlugin.scala [hunk 2]
```diff
@@ -17,10 +17,20 @@ object ProjectPlugin extends AutoPlugin {
   override def requires: Plugins = plugins.JvmPlugin && HeaderPlugin && OrgPoliciesPlugin
 
   object autoImport {
-    lazy val http4sV = "0.15.7a"
+
+    object V {
+      lazy val http4s      = "0.20.13"
+      lazy val circe       = "0.12.3"
+      lazy val log4s       = "1.7.0"
+      lazy val scalatest   = "3.0.8"
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: 3.1.0

File: project/ProjectPlugin.scala [hunk 3]
```diff
@@ -17,10 +17,20 @@ object ProjectPlugin extends AutoPlugin {
...
+      lazy val roshttp     = "2.2.4"
+      lazy val slf4jSimple = "1.7.28"
+      lazy val jwtCore     = "4.0.0"
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: 4.2.0

File: project/ProjectPlugin.scala [hunk 4]
```diff
@@ -17,10 +17,20 @@ object ProjectPlugin extends AutoPlugin {
...
 
   object autoImport {
-    lazy val http4sV = "0.15.7a"
+
+    object V {
+      lazy val http4s      = "0.20.13"
+      lazy val circe       = "0.12.3"
+      lazy val log4s       = "1.7.0"
+      lazy val scalatest   = "3.0.8"
+      lazy val roshttp     = "2.2.4"
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: We probably can remove this dependency

File: project/ProjectPlugin.scala [hunk 5]
```diff
@@ -59,11 +69,49 @@ object ProjectPlugin extends AutoPlugin {
...
 
+    lazy val serverHttpDependencies = Seq(
+      libraryDependencies ++= Seq(
+        %%("circe-core", V.circe),
+        %%("circe-generic", V.circe),
+        %("slf4j-simple", V.slf4jSimple),
+        %%("http4s-dsl", V.http4s),
+        %%("http4s-blaze-server", V.http4s),
+        %%("http4s-circe", V.http4s),
+        %("config"),
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: Could we use the versions explicitely? `V.config` -> 1.4.0

File: build.sbt [hunk 6]
```diff
@@ -1,68 +1,25 @@
...
-lazy val `evaluator-shared` = (crossProject in file("shared"))
+lazy val `evaluator-shared` = (project in file("shared"))
   .enablePlugins(AutomateHeaderPlugin)
   .settings(name := "evaluator-shared")
 
-lazy val `evaluator-shared-jvm` = `evaluator-shared`.jvm
-lazy val `evaluator-shared-js`  = `evaluator-shared`.js
-
-lazy val `evaluator-client` = (crossProject in file("client"))
+lazy val `evaluator-client` = (project in file("client"))
```
Comment (juanpedromoreno) [inline build.sbt]: I guess we can remove this module since it was moved to the scala-exercises server

File: .travis.yml [hunk 7]
```diff
@@ -30,7 +29,7 @@ after_success:
      sbt ++$TRAVIS_SCALA_VERSION dockerBuildAndPush;
      sbt ++$TRAVIS_SCALA_VERSION smoketests/test;
   fi
-- if [ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_PULL_REQUEST" = "false" -a "$TRAVIS_SCALA_VERSION" = "2.12.3" ]; then
+- if [ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_PULL_REQUEST" = "false" -a "$TRAVIS_SCALA_VERSION" = "2.12.10" ]; then
```
Comment (juanpedromoreno) [inline .travis.yml]: The branch for Scala 2.11 could be removed

File: build.sbt [hunk 8]
```diff
@@ -1,68 +1,17 @@
...
-  .settings(noPublishSettings: _*)
-  .aggregate(
-    `evaluator-server`,
-    `evaluator-shared-jvm`,
-    `evaluator-shared-js`,
-    `evaluator-client-jvm`,
-    `evaluator-client-js`)
-
-lazy val `evaluator-shared` = (crossProject in file("shared"))
+lazy val `evaluator-shared` = (project in file("shared"))
```
Comment (juanpedromoreno) [inline build.sbt]: Do we need this shared module? I'm guessing that we can convert this project in a mono-module one (everything under root)

File: client/src/main/scala/org/scalaexercises/evaluator/EvaluatorClient.scala [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+package org.scalaexercises.evaluator
+
+import cats.effect.{ConcurrentEffect, Resource}
+import org.http4s.client.Client
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.scalaexercises.evaluator.service.{HttpClientHandler, HttpClientService}
+
+import scala.concurrent.ExecutionContext
+
+object EvaluatorClient {
```
Comment (juanpedromoreno) [inline client/src/main/scala/org/scalaexercises/evaluator/EvaluatorClient.scala]: Can the `client` folder be removed completely?

File: server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala [hunk 10]
```diff
@@ -153,10 +152,12 @@ class EvalEndpointSpec extends FunSpec with Matchers {
...
-            dependencies = List(Dependency(
-              "org.scala-exercises",
-              "exercises-stdlib_2.11",
-              exercisesVersion)) ++ scalaDependencies(Scala211)
+            dependencies = List(
+              Dependency(
+                "org.scala-exercises",
+                "exercises-stdlib_2.12",
+                exercisesVersion,
+                Some(List(Exclusion("io.monix", "monix_2.11"))))) ++ scalaDependencies(Scala211)
```
Comment (juanpedromoreno) [inline server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala]: Probably we don't need these exclusions?

File: build.sbt (hunk 8 above)
Comment (kiroco12) [reply build.sbt]: I can move types.scala in server project, but smoketests is an independent project.

Comment (juanpedromoreno) [reply build.sbt]: Sounds good, we could have those two modules: server and smoketests 👍

File: build.sbt [hunk 11]
```diff
@@ -1,92 +1,34 @@
...
-    )
+    smoketestDependencies
   )
   .settings(buildInfoSettings: _*)
 
+lazy val root = (project in file("."))
+  .settings(mainClass in Universal := Some("org.scalaexercises.evaluator.EvaluatorServer"))
+  .settings(stage := (stage in Universal in `evaluator-server`).value)
+  .settings(noPublishSettings: _*)
+  .aggregate(`evaluator-server`)
```
Comment (juanpedromoreno) [inline build.sbt]: We should aggregate the smoke tests to also compile from the root module

File: project/build.properties [hunk 12]
```diff
@@ -1 +1 @@
-sbt.version=0.13.12
\ No newline at end of file
+sbt.version=1.2.8
```
Comment (juanpedromoreno) [inline project/build.properties]: Could we upgrade to sbt 1.3.4?

File: build.sbt (hunk 11 above)
Comment (kiroco12) [reply build.sbt]: In master the smoketests project wasn't aggregated to root. Maybe can be compiled, but the execution of those tests requires deployment

**Pull request merged**